archaius2-persisted2/src/main/java/com/netflix/archaius/persisted2/Persisted2ConfigProvider.java

```diff
@@ -0,0 +1,117 @@
+package com.netflix.archaius.persisted2;
+
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import com.netflix.archaius.Config;
+import com.netflix.archaius.config.PollingDynamicConfig;
+import com.netflix.archaius.config.polling.FixedPollingStrategy;
+import com.netflix.archaius.persisted2.loader.HTTPStreamLoader;
+
+/**
+ * Provider that sets up a Config that is a client to a Persisted2 service.
+ * Once injected the Config will poll the service for updated on a configurable
+ * interval.  Note that injection of this Config will block until the first
+ * set of properties has been fetched from teh remote service.
+ * 
+ * The provider must be bound to a specific config layer within the override
+ * hierarchy.
+ * 
+ * For example, Persisted2Config may boudn to the OverrideLayer like this,
+ * <pre>
+ *   final Persisted2ClientConfig config = new DefaultPersisted2ClientConfig()
+ *          .withServiceUrl("http://persiste2serviceurl")
+ *          .withQueryScope("env",    "test")
+ *          .withQueryScope("region", "us-east-1")
+ *          .withScope("env",    "test")
+ *          .withScope("region", "us-east-1")
+ *          .withScope("asg",    "app-v0001")
+ *          .withPrioritizedScopes("env", "region", "stack", "serverId");
+ *          
+ *   Guice.createInjector(
+ *      Modules
+ *          .override(new ArchaiusModule())
+ *          .with(new AbstractModule() {
+ *              @Override
+ *              protected void configure() {
+ *                  bind(Persisted2ClientConfig.class).toInstance(config);
+ *                  bind(Config.class).annotatedWith(OverrideLayer.class).toProvider(Persisted2ConfigProvider.class).in(Scopes.SINGLETON);
+ *              }
+ *          })
+ *      )
+ * </pre>
+ * 
+ * @author elandau
+ *
+ */
+public class Persisted2ConfigProvider implements Provider<Config> {
+    private final String              url;
+    private final Persisted2ClientConfig config;
+    
+    @Inject
+    public Persisted2ConfigProvider(Persisted2ClientConfig config) throws Exception {
+        this.url = new StringBuilder()
+            .append(config.getServiceUrl())
+            .append("?skipPropsWithExtraScopes=").append(config.getSkipPropsWithExtraScopes())
+            .append("&filter=").append(URLEncoder.encode(getFilterString(config.getQueryScopes()), "UTF-8"))
+            .toString();
+        this.config = config;
+    }
+    
+    public static String getFilterString(Map<String, List<String>> scopes) {
+        StringBuilder sb = new StringBuilder();
+        for (Entry<String, List<String>> scope : scopes.entrySet()) {
+            if (sb.length() > 0 && !scope.getValue().isEmpty()) {
+                sb.append(" and ");
+            }
```

Comment:
If I'm reading this right it would add empty parens if one of the scopes is empty, e.g.:

```json
{
  "scope1": ["foo"],
  "scope2": [],
  "scope3": ["bar"]
}
```

Would lead to a query like:

```
(scope1='foo') () and (scope3='bar')
```